apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/add-partner-form.client.tsx

```diff
@@ -0,0 +1,138 @@
+"use client";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { z } from "zod";
+import type { Ecosystem } from "../../../types";
+import { useAddPartner } from "../../hooks/use-add-partner";
+
+const formSchema = z.object({
+  name: z.string().min(1, {
+    message: "Partner name is required",
+  }),
+  domains: z
+    .string()
+    .transform((value) => value.split(/,| /))
+    .pipe(z.string().array())
+    .optional(),
+  bundleIds: z
+    .string()
+    .transform((value) => value.split(/,| /))
+    .pipe(z.string().array())
+    .optional(),
+});
+
+export function AddPartnerForm({ ecosystem }: { ecosystem: Ecosystem }) {
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+  });
+
+  const { addPartner, isLoading } = useAddPartner({
+    onSuccess: () => {
+      form.reset();
+    },
+    onError: (error) => {
+      const message =
+        error instanceof Error
+          ? error.message
+          : "Failed to add ecosystem partner";
+      toast.error(message);
+    },
+  });
+
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit((values) =>
+          addPartner({
+            ecosystem,
+            name: values.name,
+            allowlistedDomains: values.domains ?? [],
+            allowlistedBundleIds: values.bundleIds ?? [],
+            permissions: "PROMPT_USER_V1",
+          }),
+        )}
+        className="flex flex-col gap-2 md:flex-row"
+      >
+        <div className="grid gap-2 md:grid-cols-11 grow">
+          <FormField
+            control={form.control}
+            name="name"
+            defaultValue="" // Note: you *must* provide a default value here or the field won't reset
+            render={({ field }) => (
+              <FormItem className="col-span-4 md:col-span-3">
+                <FormControl>
+                  <Input
+                    placeholder="App name"
+                    {...field}
+                    value={field.value}
+                  />
+                </FormControl>
+                <FormMessage />
+              </FormItem>
+            )}
+          />
+          <FormField
+            control={form.control}
+            name="domains"
+            defaultValue={[]} // Note: you *must* provide a default value here or the field won't reset
```

Comment:
is the default value truly an array? I thought this was a string input with , separation?